### ISSUE TITLE
sbt community repo (repo.scala-sbt.org)

### ISSUE THREAD

**Issue by eed3si9n** [sbt/sbt#6294]
JFrog announced that they are sunsetting Bintray starting February 28th, 2021.

https://jfrog.com/blog/into-the-sunset-bintray-jcenter-gocenter-and-chartcenter/

**Update 2021-04-28**:

Please read [Bintray to JFrog Artifactory migration status and sbt 1.5.1](https://eed3si9n.com/bintray-to-jfrog-artifactory-migration-status-and-sbt-1.5.1).
There's now a cloud-hosted Artifactory instance licensed to Scala Center. With the release of sbt 1.5.1, I think we can say that the migration is done.

**Update 2021-04-12**:

- <s>Currently sbt community repo is down due to Bintray's scheduled brown out until 2021-04-12 21:00 UTC (or 5pm Eastern) - https://status.bintray.com/</s>

## The plan

1. <s>Existing sbt plugins and Linux installers will be migrated to a new Artifactory instance sponsored by JFrog. Currently there's no ETA on this, but it should be completed before May 1st, 2021.</s> Done.
2. repo.scala-sbt.org will re-point its redirection to that Artifactory URL. Please keep using repo.scala-sbt.org since we might have to move again in the future (some might remember it used to point to Artifactory before it was Bintray).

## Dear plugin authors

_Existing_ versions of plugins released to sbt Community Repo are migrated to Artifactory. So the current builds will hopefully continue to work without making changes.

Let's use this as an opportunity to wean ourselves from the Ivy repo. For _future_ plugin releases, please migrate to Sonatype OSS (Maven Central).

**Comment by eed3si9n** [comment]
Scala Center, its Advisory Board members, VirtusLab, Lightbend, and JFrog are discussing the next steps. Likely we will try to figure out a way to keep the repository going in its current shape, just not implemented as Bintray. Hopefully more detail will be forthcoming next week.

**Comment by metasim** [comment]
@eed3si9n Many many thanks for keeping the community in the loop. Very much appreciated.

**Comment by ScalaWilliam** [comment]
Would there be any plans to publish to Maven Central?

I understand repo.scala-sbt.org is a separate repository of its own and often needs to be enabled explicitly by enterprises which is a tremendous task from the aspect of red tape (think: 6 months of meetings to get it approved -- is enough time to Kotlin to jump in and take the spot) - we could have a far greater adoption of Scala if all the necessary tools were available immediately off Maven Central. Please correct if my understanding is wrong.

It is a good opportunity to do this now especially that JFrog are sunsetting some of their services.

**Comment by eed3si9n** [comment]
We're forming Bintray Solution Task Force at Scala Center to discuss the next steps.

As Scala community (open source user, corporate users, etc using Scala) I think the 3 pressing priorities are:
- Continuity of build semantics (existing builds continue to work without making changes) "read problem"
- Maintaining secure custody (migration and confusion can create cover for supply chain attacks) "security problem"
- sbt installers for Linux distributions (deb, and rpm files are hosted on Bintray) "Linux distro problem"

Secondary issues are
- Allowing plugins to publish "plugin write problem"
- Automatic publishing of sbt nightly builds "nightlies problem"

In terms of the deadlines, we have until May 1st to resolve "read" and "security" problem, whereas "Linux distro problem" and other write issues will be disrupted on February 28th.

## read problem

### batch migration to Maven Central

A few people have suggested the idea of wholesale migration to Maven Central, but I don't think that would work for multiple reasons:
1. sbt plugin releases is an Ivy-layout repository.
2. Many of the existing artifacts do not meet the Sonatype OSS requirements (GPG signing, domain ownership, POM metadata like "description" etc)

### Artifactory

On the other hand, JFrog has a repository manager Artifactory that is open source, and perfectly capable of hosting Ivy-layout repositories (and Linux distributions). There's also a hosted version of Artifactory called Artifactory Pro. Fortunately, there's [Artifactory OSS Sponsorship Program](https://jfrog.com/blog/jfrog-artifactory-saas-google-cloud-platform-gcp/) sponsored by JFrog and Google, and last I heard sbt project has been approved for it. This likely means that we won't have to pay for the GB/month network fees. We will find out more about the details.

## security problem

We will find out more, but if the migration from Bintray to Artifactory could happen within, this also minimizes the security concern regarding custody transfer at least in the short-term.

## write problem

Among the Task Force members the prevalent opinion is that we should use this as an opportunity to wean ourselves off of the Ivy layout and migrate to Maven Central, regardless of what we do on the read side. If plugins authors want to start on that they can.

For core plugins that I help maintain (but didn't start), I've claimed `com.github.sbt` on Maven Central, and set up automatic publishing using [sbt-ci-release](https://github.com/olafurpg/sbt-ci-release).

**Comment by ScalaWilliam** [comment]
**Regarding Maven Central**:
Using SBT behind the firewall has been challenging because an extra repo, in a different format, needs a lot of organizational manouvering, easily a 3-9 month process, stopping Scala from being adopted at some organizations, as SBT is the main build tool and the quality of other build tools for Scala is not great.

We have a lot to lose by not at least having the essential parts of SBT (which IMO is very crucial to Scala) on Maven Central. I hope authors can publish their plug-ins to Maven Central so that SBT is easily accessible for everybody, because Maven Central is already supported at most places.

**Comment by eed3si9n** [comment]
I feel like we are talking past each other. I am not disagreeing that gradually plugin authors can move the next versions of the plugins to Maven Central. I've already started doing so for sbt-pgp for example.

We have 700 **existing** plugins and numerous **existing** builds (including its git history). For the sake of build continuity and prevention of supply chain attack, Scala Center and friends should provide a repository that functions exactly like what we have today for read before May 1st is what I'm saying.

**Comment by ScalaWilliam** [comment]
Thank you for clarifying @eed3si9n, definitely in agreement over existing builds, not to break compatibility.
I had misunderstood and missed part of what you meant in your previous comment - my apologies.

This indeed answers the question of publishing plugins to Maven Central -- https://github.com/olafurpg/sbt-ci-release#how-do-i-publish-sbt-plugins -- if correctly understood, this would mean that consuming them would also be transparent, or is there still a need to configure a custom ~/.sbt/repositories file to consume these plugins?

**Comment by SethTisue** [comment]
@ScalaWilliam I’m glad you brought up these concerns, even just to double check on them. Ultimately, once widely used plugins have published new versions to Maven Central, adoption of Scala at the kind of locked-down shops you describe will be easier, and that’s a very good thing — a silver lining to this cloud.

**Comment by sjrd** [comment]
> if correctly understood, this would mean that consuming them would also be transparent, or is there still a need to configure a custom ~/.sbt/repositories file to consume these plugins?

It's transparent. In fact, you may already be using sbt plugins published on Maven Central, without knowing it. sbt-scalajs 1.5.0 is one example.

**Comment by sanjivsahayamrea** [comment]
Are there plans to move any of the [existing plugins](https://dl.bintray.com/sbt/sbt-plugin-releases/) hosted on Bintray (older versions etc) to Sonatype as well?  https://github.com/foundweekends/giter8/issues/513 affects only new publishes right?

**Comment by SethTisue** [comment]
@sanjivsahayamrea that's already been answered by Eugene at https://github.com/sbt/sbt/issues/6294#issuecomment-782733993 (TL;DR: no, for good reasons)

**Comment by sanjivsahayamrea** [comment]
Thanks @SethTisue , I didn't see that 👍

**Comment by sanjivsahayamrea** [comment]
I'm sorry if this is not the issue to ask this question on. I'm happy to move it to another issue if that's preferable. 

The following default SBT resolvers that are pointing to Bintray at the moment:

| Resolver | Default Url | Redirected URL |
| -------- | --- | ---- |
| typesafeIvyRepo | https://repo.typesafe.com/typesafe/ivy-releases | https://dl.bintray.com/typesafe/ivy-releases/ |
typesafeRepo    | https://repo.typesafe.com/typesafe/releases | https://dl.bintray.com/typesafe/maven-releases |
sbtPluginRepo   | https://repo.scala-sbt.org/scalasbt/sbt-plugin-releases | https://dl.bintray.com/sbt/sbt-plugin-releases  |
jcenterRepo  |  https://jcenter.bintray.com/ | https://jcenter.bintray.com/ |


Will these URLs get redirected to Maven Central or some other repo before the 1st of May?

**Comment by eed3si9n** [comment]
Scala Center + friends will take care of sbt plugin repository, and hopefully before May 1st.

**Comment by ak0rz** [comment]
jfyi: every sbt build is broken today because of Bintray brown-out session https://status.bintray.com/

**Comment by sjrd** [comment]
Yes, I guess that was to be expected.

We are still working on migrating the sbt plugins published on Bintray to a new Artifactory instance. We are doing everything in our power to finish that before May 1st.

**Comment by ak0rz** [comment]
> Yes, I guess that was to be expected.
> 
> We are still working on migrating the sbt plugins published on Bintray to a new Artifactory instance. We are doing everything in our power to finish that before May 1st.

Okay, looks like I'll schedule a day-off on April 26th. :D

**Comment by eed3si9n** [comment]
JFrog has generously granted sbt project their open source sponsorship program which lets us use a cloud-hosted Artifactory instance with CDN, and we have chosen Scala Center as the new owner for it

https://scala.jfrog.io will be our new home for the sbt plugins

<img width="872" alt="JFrog-Artifactory" src="https://user-images.githubusercontent.com/184683/114418691-c3eadb80-9b80-11eb-942e-058d31e2f055.png">

**Comment by cleliofs** [comment]
> JFrog has generously granted sbt project their open source sponsorship program which lets us use a cloud-hosted Artifactory instance with CDN, and we have chosen Scala Center as the new owner for it
> 
> https://scala.jfrog.io will be our new home for the sbt plugins
> 

So, can we just update the existing resolver URLs to "https://scala.jfrog.io" instead?

**Comment by eed3si9n** [comment]
It's not ready yet, and when it's done https://repo.scala-sbt.org/ should work without any change as it's a redirect proxy. This the second time it moved and it might move again, so I suggest we keep doing redirect thing.

**Comment by ak0rz** [comment]
> It's not ready yet, and when it's done https://repo.scala-sbt.org/ should work without any change as it's a redirect proxy. This the second time it moved and it might move again, so I suggest we keep doing redirect thing.

Eugene, is there any ETA on that? It's a bit painful to be unable to build services even for a day. :(
Maybe we can help with smth?

**Comment by joschi** [comment]
@ya-goodfella You can use the workaround described https://github.com/aws/aws-codebuild-docker-images/issues/449#issuecomment-817884851 to make your build jobs run today until the sbt team had time to react and set everything up.

When using the workaround, you'll have to install `sbt` with an alternative installation method if your builds depend on it being preinstalled.

**Comment by ak0rz** [comment]
> @ya-goodfella You can use the workaround described [aws/aws-codebuild-docker-images#449 (comment)](https://github.com/aws/aws-codebuild-docker-images/issues/449#issuecomment-817884851) to make your build jobs run today until the sbt team had time to react and set everything up.

Well, it's not about APT repositories and installing sbt.
It's about resolving sbt plugins at startup.

**Comment by eed3si9n** [comment]
> Eugene, is there any ETA on that? It's a bit painful to be unable to build services even for a day. :(
> Maybe we can help with smth?

sbt is no longer my day job, so this would have to wait. I don't really have a ETA on when I'd run the migration or how long it might take.

**Comment by SethTisue** [comment]
I put the word out about April 26 on Reddit (https://www.reddit.com/r/scala/comments/mph43t/sbt_build_failing_bintray_brownout_today_april_12/) and Twitter (https://twitter.com/SethTisue/status/1381624559296675846)

**Comment by ak0rz** [comment]
> > Eugene, is there any ETA on that? It's a bit painful to be unable to build services even for a day. :(
> 
> > Maybe we can help with smth?
> 
> 
> 
> sbt is no longer my day job, so this would have to wait. I don't really have a ETA on when I'd run the migration or how long it might take.
> 
> 

Well, I'm not really pushing you, just curious about what's the plan. My apologies if it seemed so. :)

Looks like we need to:
1. Shut down artifact upload to bintray for sbt plugin releases
2. [Mirror everything from bintray to new jfrog artifactory](https://jfrog.com/knowledge-base/how-to-mirror-a-remote-repository/)
3. Create new redirects under scala-sbt domain
4. Place an updated instruction for plugin authors about how to release plugins afterwards.

Am I right?

**Comment by mcintyre94** [comment]
> It's not ready yet, and when it's done https://repo.scala-sbt.org/ should work without any change as it's a redirect proxy. This the second time it moved and it might move again, so I suggest we keep doing redirect thing.

In the docs where it uses the bintray link (https://www.scala-sbt.org/1.x/docs/Installing-sbt-on-Linux.html#Ubuntu+and+other+Debian-based+distributions)

`echo "deb https://dl.bintray.com/sbt/debian /" | sudo tee -a /etc/apt/sources.list.d/sbt.list`

Should this be the following?

`echo "deb https://repo.scala-sbt.org/debian /" | sudo tee -a /etc/apt/sources.list.d/sbt.list`

I'm happy to PR a docs update if this is correct, but while https://repo.scala-sbt.org/ is pointing at bintray and they're having their brownout I can't verify what the correct URL is myself.

**Comment by kuberlog** [comment]
The ball was dropped here. Is lightbend even profitable? It seems like they don't care about outages.

**Comment by SethTisue** [comment]
@eed3si9n is sbt 0.13.x part of the rescue plan, or only sbt 1.x? (should I open a separate ticket just for 0.13 discussion?)

**Comment by Jasper-M** [comment]
> https://scala.jfrog.io will be our new home for the sbt plugins

Will this also be the new home for other Scala artifacts hosted on bintray such as the ones in bintray.com/typesafe?

**Comment by SethTisue** [comment]
@Jasper-M Lightbend IT is in the process of securing new hosting for repo.lightbend.com and repo.typesafe.com; they expect to be done before May 1, and those resolvers will continue functioning for users as before, via updated redirects

but that's just Lightbend/Typesafe stuff. lord knows how many other Scala OSS projects were published to Bintray under other accounts. it's not feasible for the Center (or for member companies such as Lightbend) to identify and re-host all of them

**Comment by eed3si9n** [comment]
@SethTisue 

> @eed3si9n is sbt 0.13.x part of the rescue plan, or only sbt 1.x? (should I open a separate ticket just for 0.13 discussion?)

If I recall correctly the initial plan for Lightbend was to migrate all repo.typesafe.com, but given the complexity involved in translating Ivy layout to Maven, we should piggy back https://bintray.com/typesafe/ivy-releases (including sbt 0.13 core artifacts) to new home at least for the time being to ensure continuity of the build. /cc @JustinPihony @sebastian-alfers

**Comment by mcintyre94** [comment]
FYI: If you have a URL like `http://dl.bintray.com/sbt/debian/sbt-${SBT_VERSION}.deb` (in our case in a Dockerfile), then you can update it to the redirect proxy by changing it to `https://repo.scala-sbt.org/scalasbt/debian/sbt-${SBT_VERSION}.deb`.

**Comment by joseph-whiting** [comment]
Can I ask what the current projection is for April 26th? The brown-out was a bit painful for our dev team, so would be helpful to know whether or not we should be thinking about workarounds from our end to prevent disruption on the 26th.

Thanks for all your work!

**Comment by eed3si9n** [comment]
@joseph-whiting 

We will try to complete the migration by the next brown out.

**Comment by eed3si9n** [comment]
I'm not going to share the URL since we should continue to use https://repo.scala-sbt.org/, but I've migrated sbt/sbt-plugin-releases, sbt/debian, sbt/rpm, and typesafe/ivy-releases into Artifactory last night.

**Comment by ak0rz** [comment]
> I'm not going to share the URL since we should continue to use https://repo.scala-sbt.org/, but I've migrated sbt/sbt-plugin-releases, sbt/debian, sbt/rpm, and typesafe/ivy-releases into Artifactory last night.

That's awesome!

Really appreciate it.

Thank you so much, Eugene!

**Comment by JustinPihony** [comment]
To be safe, it should probably also be noted for anybody who is watching this - if you push a new release to bintray after Eugene's migration then it will NOT be there after the sunset on May 1. Please do NOT publish any new releases to Bintray - follow Eugene's direction instead:

> For future plugin releases, please migrate to Sonatype OSS (Maven Central).

**Comment by ScalaWilliam** [comment]
> To be safe, it should probably also be noted for anybody who is watching this - if you push a new release to bintray after Eugene's migration then it will NOT be there after the sunset on May 1. Please do NOT publish any new releases to Bintray - follow Eugene's direction instead:
> 
> > For future plugin releases, please migrate to Sonatype OSS (Maven Central).

Yes please. We need more plugins on Maven Central, life is difficult for those in corporates ;-)

**Comment by sjrd** [comment]
It hasn't been possible to push new releases to Bintray since March 1st, IIRC the original announcement by JFrog.

**Comment by JustinPihony** [comment]
> It hasn't been possible to push new releases to Bintray since March 1st, IIRC the original announcement by JFrog.

No, that was for JCenter, GoCenter, and ChartCenter - but Bintray will go both ways until it's sunset.

**Comment by JustinPihony** [comment]
The new resolver redirects are active. So, if you are using repo.scala-sbt.org then you should automatically start resolving from the new location. And if you aren't then you should change to repo.scala-sbt.org before the 26th brown out to avoid downtime/confusion.

**Comment by eed3si9n** [comment]
Thanks Justin! Everyone should test their favorite build using the following:

```
env COURSIER_CACHE=/tmp/cache0 sbt
```

**Comment by eed3si9n** [comment]
I'm going to close this as resolved for now.

**Comment by dimisjim** [comment]
@JustinPihony 

How can someone, without a phd in sbt, quickly inspect this in his own project(s), and be certain that every project that should use the `repo.scala-sbt.org`, is actually using it?

I have tried https://stackoverflow.com/a/41094190/3263374 but the output is completely unreadable

Also I have tried to run `sbt "show fullResolvers" | grep repo.scala-sbt.org` but I don't get anything back. Does this mean that `repo.scala-sbt.org` is not used at all?

@eed3si9n Could you give a few concrete examples on how to be certain that `repo.scala-sbt.org` is being used?

**Comment by osumatu** [comment]
> The new resolver redirects are active. So, if you are using repo.scala-sbt.org then you should automatically start resolving from the new location. And if you aren't then you should change to repo.scala-sbt.org before the 26th brown out to avoid downtime/confusion.

Just one small addition:  We have just realized in the company, indeed redirects are active but looks like it creates an another proxy call and we have only the old link in the white list so we started getting HTTP 403 error for our builds. We are adding the new link also to the white list.

**Comment by jrudolph** [comment]
> > > For future plugin releases, please migrate to Sonatype OSS (Maven Central).
> 
> Yes please. We need more plugins on Maven Central, life is difficult for those in corporates ;-)

In that case, would it be possible to change sbt's patterns for sbt plugins to follow the usual Maven rules? Right now, `_<Scala version>_<sbt version>` is appended to the name (the directory in a maven repo) but not to all the artifact names. E.g.

```diff
-https://repo1.maven.org/maven2/net/virtual-void/sbt-dependency-graph_2.12_1.0/0.10.0-RC1/sbt-dependency-graph-0.10.0-RC1.jar
+https://repo1.maven.org/maven2/net/virtual-void/sbt-dependency-graph_2.12_1.0/0.10.0-RC1/sbt-dependency-graph_2.12_1.0-0.10.0-RC1.jar
```

Why is that important? oss.sonatype.org still accepts those artifacts but e.g. there are no download statistics available for artifacts named like this. Also other tools like Maven proxies sometimes choke on weirdly-named artifacts because they may be stricter or they interpret file names and apply some logic that may fail with sbt's pattern.

**Comment by jrudolph** [comment]
I just had the sbt launcher script fail on travis because the md5 files for the launcher jar seem now to be missing.

See https://travis-ci.com/github/akka/akka-http/jobs/499459555

```
$ sbt -jvm-opts .jvmopts-travis "$CMD"

Detected sbt version 1.4.7

Downloading sbt launcher for 1.4.7:

  From  https://repo.scala-sbt.org/scalasbt/maven-releases/org/scala-sbt/sbt-launch/1.4.7/sbt-launch-1.4.7.jar

    To  /home/travis/.sbt/launchers/1.4.7/sbt-launch.jar

Downloading sbt launcher 1.4.7 md5 hash:

  From  https://repo.scala-sbt.org/scalasbt/maven-releases/org/scala-sbt/sbt-launch/1.4.7/sbt-launch-1.4.7.jar.md5

    To  /home/travis/.sbt/launchers/1.4.7/sbt-launch.jar.md5

cat: /home/travis/.sbt/launchers/1.4.7/sbt-launch.jar.md5: No such file or directory

md5sum: 'standard input': no properly formatted MD5 checksum lines found

Checksum does not match

Could not download and verify the launcher. Obtain the jar manually and place it at /home/travis/.sbt/launchers/1.4.7/sbt-launch.jar
```

**Comment by jrudolph** [comment]
> I just had the sbt launcher script fail on travis because the md5 files for the launcher jar seem now to be missing.

This is probably a version of https://github.com/paulp/sbt-extras that does the check which comes predefined with Travis if `language: scala` is set.

**Comment by eed3si9n** [comment]
@jrudolph That's right. JFrog's migration script removed all `*.md5` files (because Artifactory has the information in HTTP HEAD?) and unofficial runners seem to be affected.

**Comment by jrudolph** [comment]
> @jrudolph That's right. JFrog's migration script removed all `*.md5` files (because Artifactory has the information in HTTP HEAD?) and unofficial runners seem to be affected.

Unofficial but de facto standard, so I guess it needs to be fixed anyways...

**Comment by jrudolph** [comment]
FTR, adding the option `-sbt-launch-repo https://repo1.maven.org/maven2` to sbt in travis helps with that issue.

**Comment by eed3si9n** [comment]
@dimisjim  

> Could you give a few concrete examples on how to be certain that repo.scala-sbt.org is being used?

For sbt 1.x builds repo.scala-sbt.org is used only for plugins. If you move around the Coursier and Ivy cache directories like

```bash
env COURSIER_CACHE=/tmp/cache0 sbt -Dsbt.ivy.home=/tmp/cache1
```

you should see sbt plugins load from new repo.scala-sbt.org. You can see the resolvers in metabuild as follows:


```
sbt:sbtRoot> reload plugins
....
sbt:project> show fullResolvers
[info] * Raw(ProjectResolver(inter-project, mapped: ))
[info] * public: https://repo1.maven.org/maven2/
[info] * URLRepository(sbt-plugin-releases, Patterns(ivyPatterns=Vector(https://repo.scala-sbt.org/scalasbt/sbt-plugin-releases/[organisation]/[module]/(scala_[scalaVersion]/)(sbt_[sbtVersion]/)([branch]/)[revision]/[type]s/[artifact](-[classifier]).[ext]), artifactPatterns=Vector(https://repo.scala-sbt.org/scalasbt/sbt-plugin-releases/[organisation]/[module]/(scala_[scalaVersion]/)(sbt_[sbtVersion]/)([branch]/)[revision]/[type]s/[artifact](-[classifier]).[ext]), isMavenCompatible=false, descriptorOptional=false, skipConsistencyCheck=false), false)
[info] * URLRepository(typesafe-alt-ivy-releases, Patterns(ivyPatterns=Vector(https://repo.typesafe.com/typesafe/ivy-releases/[organisation]/[module]/(scala_[scalaVersion]/)(sbt_[sbtVersion]/)([branch]/)[revision]/[type]s/[artifact](-[classifier]).[ext]), artifactPatterns=Vector(https://repo.typesafe.com/typesafe/ivy-releases/[organisation]/[module]/(scala_[scalaVersion]/)(sbt_[sbtVersion]/)([branch]/)[revision]/[type]s/[artifact](-[classifier]).[ext]), isMavenCompatible=false, descriptorOptional=false, skipConsistencyCheck=false), false)
[info] * FileRepository(local, Patterns(ivyPatterns=Vector(${ivy.home}/local/[organisation]/[module]/(scala_[scalaVersion]/)(sbt_[sbtVersion]/)([branch]/)[revision]/[type]s/[artifact](-[classifier]).[ext]), artifactPatterns=Vector(${ivy.home}/local/[organisation]/[module]/(scala_[scalaVersion]/)(sbt_[sbtVersion]/)([branch]/)[revision]/[type]s/[artifact](-[classifier]).[ext]), isMavenCompatible=false, descriptorOptional=false, skipConsistencyCheck=false), FileConfiguration(true, None))
[info] * FileRepository(local-preloaded-ivy, Patterns(ivyPatterns=Vector(/Users/eed3si9n/.sbt/preloaded/[organization]/[module]/[revision]/[type]s/[artifact](-[classifier]).[ext]), artifactPatterns=Vector(/Users/eed3si9n/.sbt/preloaded/[organization]/[module]/[revision]/[type]s/[artifact](-[classifier]).[ext]), isMavenCompatible=false, descriptorOptional=false, skipConsistencyCheck=false), FileConfiguration(true, None))
[info] * local-preloaded: file:////Users/eed3si9n/.sbt/preloaded/
[success] Total time: 0 s, completed Apr 19, 2021 10:19:58 AM
sbt:project> reload return
```

**Comment by dimisjim** [comment]
@eed3si9n ~I still don't see any `scala-sbt.org` in the output.~

~Should I add something in the resolvers path in `build.sbt`?~

EDIT: After doing it interactively, I now can see it.

**Comment by bjaglin** [comment]
Thanks @eed3si9n & @JustinPihony  for the smooth migration!

Regarding the [sbt-extras](https://github.com/paulp/sbt-extras/) regression, my understanding is that any script prior to https://github.com/paulp/sbt-extras/pull/305 running on a clean `~/.sbt` is now broken (or require `-sbt-launch-repo https://repo1.maven.org/maven2`). Therefore, it does not seem to be related to the lack of md5 files in Artifactory but rather to the removal of https://repo.scala-sbt.org/scalasbt/maven-releases.

Could we then consider keeping https://repo.scala-sbt.org/scalasbt/maven-releases as a redirection to https://repo1.maven.org/maven2? It seems to be used [quite widely](https://github.com/search?q=scalasbt%2Fmaven-releases&type=code).

**Comment by JustinPihony** [comment]
Sorry for not commenting earlier, but I actually went ahead and did this yesterday. Although upon testing it I realized I made it a `rewrite`, but forgot the `redirect` - I have now fixed it and it should be working as expected.

**Comment by muuki88** [comment]
Hi there ☺️

Thanks for the fast response on all this.

Is there any guidance on how plugin authors should migrate to maven central?

Should every maintainer get credentials from sonatype? And if yes, how do we ensure if someone is eligible? Coordinates should be as specific as possible, I guess?

I'll try to start next week with migrating sbt-native-packager. I'm happy for any blog posts or repositories already publishing SBT plugins to maven central 🥰

**Comment by sjrd** [comment]
I'll do my best to write something up on Monday.

**Comment by eed3si9n** [comment]
Given that there's a brown out coming up today, I figured I'll do a blog post sooner :) 
https://eed3si9n.com/bintray-to-jfrog-artifactory-migration-status-and-sbt-1.5.1

**Comment by mkurz** [comment]
@eed3si9n I can't upgrade sbt on my Ubuntu using `sudo apt-get update && sudo apt-get dist-upgrade`, I think there is a problem:
`sbt-1.5.1.deb ` exists in https://repo.scala-sbt.org/scalasbt/debian, however the latest version the `Packages` file contains is 1.4.9:
```sh
$ curl -L https://repo.scala-sbt.org/scalasbt/debian/Packages
...
<older releases>
...

Source: sbt
Package: sbt
Version: 1.4.9
Section: java
Priority: optional
Architecture: all
Installed-Size: 1411
Depends: bash (>= 3.2)
Recommends: git
Maintainer: Eugene Yokota <eugene.yokota@lightbend.com>
Description: sbt, the interactive build tool
 This script provides a native way to run sbt,
  a build tool for Scala and more.
Filename: sbt-1.4.9.deb
SHA1: d3e4a8d5af69732de549a8477321d926c3d47b4b
SHA256: d0bbca96063ca14ed9d9c0763fe06ca10dac59200987107397e2a15d9776d8c3
Size: 1233694
```

**Comment by WayneWang12** [comment]
> Given that there's a brown out coming up today, I figured I'll do a blog post sooner :)
> https://eed3si9n.com/bintray-to-jfrog-artifactory-migration-status-and-sbt-1.5.1

I've check the repository finding that there aren't any plugins of sbt 1.x.

**Comment by SethTisue** [comment]
> I've check the repository finding that there aren't any plugins of sbt 1.x.

Can you be extremely specific about exactly what you're looking for and exactly what you're seeing, please?

**Comment by eed3si9n** [comment]
@mkurz I haven't had chance to update https://www.scala-sbt.org/, but per my blog post:

- As of April 26th, Debian package will be published to `deb https://repo.scala-sbt.org/scalasbt/debian all main` on Artsy. The older releases are hosted on `deb https://repo.scala-sbt.org/scalasbt/debian /`.

```bash
echo "deb https://repo.scala-sbt.org/scalasbt/debian all main" | sudo tee /etc/apt/sources.list.d/sbt.list
echo "deb https://repo.scala-sbt.org/scalasbt/debian /" | sudo tee /etc/apt/sources.list.d/sbt_old.list
curl -sL "https://keyserver.ubuntu.com/pks/lookup?op=get&search=0x2EE0EA64E40A89B84B2DF73499E82A75642AC823" | sudo apt-key add
sudo apt-get update
sudo apt-get install sbt
```

**Comment by WayneWang12** [comment]
> > I've check the repository finding that there aren't any plugins of sbt 1.x.
> 
> Can you be extremely specific about exactly what you're looking for and exactly what you're seeing, please?

here: https://scala.jfrog.io/ui/

In sbt-plugin-releases.

**Comment by eed3si9n** [comment]
@WayneWang12 

https://repo.scala-sbt.org/scalasbt/sbt-plugin-releases/com.eed3si9n/sbt-assembly/scala_2.12/sbt_1.0/0.15.0/ redirects fine.

See also
<img width="1140" alt="Artsy_by_Scala_Center" src="https://user-images.githubusercontent.com/184683/116112019-2bc31b00-a685-11eb-8228-c51dd85c0499.png">

**Comment by SethTisue** [comment]
@WayneWang12 I see lots of sbt 1.0 plugins there, for example https://scala.jfrog.io/artifactory/sbt-plugin-releases/org.scalameta/sbt-metals/scala_2.12/sbt_1.0/ , tons of others too

**Comment by WayneWang12** [comment]
maybe some of it missing new versions:
https://scala.jfrog.io/artifactory/sbt-plugin-releases/com.typesafe/sbt-native-packager:
![image](https://user-images.githubusercontent.com/6812513/116116379-ac1d6680-a6ed-11eb-87f8-706ac8739598.png)

**Comment by eed3si9n** [comment]
@WayneWang12 sbt-native-packager is now under com.typesafe.sbt organization name.

Do have any builds that is experiencing build errors?

**Comment by SethTisue** [comment]
@WayneWang12 if you are having trouble with a particular plugin, it might be better to seek help in that plugin's repo, and/or to seek help on https://gitter.im/sbt/sbt, and only comment here again if you can show hard evidence that something went wrong with the overall plugin migration.

(There are probably hundreds of people watching this ticket, so doing detailed troubleshooting back-and-forth here has a higher-than-usual cost.)

**Comment by jimm-porch** [comment]
I figured it might help to know that these fixes are definitely working for some of us.

During the last brownout a couple weeks ago, our SBT builds at Porch were all failing, so I expected some trouble with today's 2nd brownout.  But everything is working fine for us today, and we didn't have to make a single change on our side.

Thanks for all the hard work everyone. 👍

**Comment by dimisjim** [comment]
Doesn't look completely fixed to me: https://github.com/sbt/sbt/issues/6474

**Comment by WayneWang12** [comment]
> @WayneWang12 sbt-native-packager is now under com.typesafe.sbt organization name.
> 
> Do have any builds that is experiencing build errors?

Yeah, just picked some randomly from my build error.  I do mistake `com.typesafe` and `com.typesafe.sbt`. Sorry for the mislead.

**Comment by vishalpawar17** [comment]
i am getting 404 for below libraries. Can someone tell me when will be those available on respective path 

```
https://repo.scala-sbt.org/scalasbt/maven-releases/com/typesafe/sbt/sbt-ghpages_2.12_1.0/0.6.3/sbt-ghpages-0.6.3.pom
https://repo.scala-sbt.org/scalasbt/maven-releases/com/typesafe/sbt/sbt-js-engine_2.12_1.0/1.2.3/sbt-js-engine-1.2.3.pom
https://repo.scala-sbt.org/scalasbt/maven-releases/com/typesafe/sbt/sbt-native-packager_2.12_1.0/1.5.2/sbt-native-packager-1.5.2.pom
https://repo.scala-sbt.org/scalasbt/maven-releases/com/eed3si9n/sbt-buildinfo_2.12_1.0/0.10.0/sbt-buildinfo-0.10.0.pom
https://repo.scala-sbt.org/scalasbt/maven-releases/com/lightbend/sbt/sbt-javaagent_2.12_1.0/0.1.5/sbt-javaagent-0.1.5.pom
https://repo.scala-sbt.org/scalasbt/maven-releases/com/github/cb372/sbt-explicit-dependencies_2.12_1.0/0.2.10/sbt-explicit-dependencies-0.2.10.pom
https://repo.scala-sbt.org/scalasbt/maven-releases/com/typesafe/sbt/sbt-site_2.12_1.0/1.4.0/sbt-site-1.4.0.pom
https://repo.scala-sbt.org/scalasbt/maven-releases/com/dwijnand/sbt-compat_2.12_1.0/1.2.6/sbt-compat-1.2.6.pom
https://repo.scala-sbt.org/scalasbt/maven-releases/com/typesafe/sbt/sbt-twirl_2.12_1.0/1.5.1/sbt-twirl-1.5.1.pom
https://repo.scala-sbt.org/scalasbt/maven-releases/io/gatling/gatling-sbt_2.12_1.0/3.1.0/gatling-sbt-3.1.0.pom
https://repo.scala-sbt.org/scalasbt/maven-releases/com/typesafe/sbt/sbt-web_2.12_1.0/1.4.4/sbt-web-1.4.4.pom
https://repo.scala-sbt.org/scalasbt/maven-releases/com/eed3si9n/sbt-unidoc_2.12_1.0/0.4.3/sbt-unidoc-0.4.3.pom
----------------

https://repo.scala-sbt.org/scalasbt/maven-snapshots/com/typesafe/sbt/sbt-ghpages_2.12_1.0/0.6.3/sbt-ghpages-0.6.3.pom
https://repo.scala-sbt.org/scalasbt/maven-snapshots/com/typesafe/sbt/sbt-js-engine_2.12_1.0/1.2.3/sbt-js-engine-1.2.3.pom
https://repo.scala-sbt.org/scalasbt/maven-snapshots/com/typesafe/sbt/sbt-native-packager_2.12_1.0/1.5.2/sbt-native-packager-1.5.2.pom
https://repo.scala-sbt.org/scalasbt/maven-snapshots/com/eed3si9n/sbt-buildinfo_2.12_1.0/0.10.0/sbt-buildinfo-0.10.0.pom
https://repo.scala-sbt.org/scalasbt/maven-snapshots/com/lightbend/sbt/sbt-javaagent_2.12_1.0/0.1.5/sbt-javaagent-0.1.5.pom
https://repo.scala-sbt.org/scalasbt/maven-snapshots/com/github/cb372/sbt-explicit-dependencies_2.12_1.0/0.2.10/sbt-explicit-dependencies-0.2.10.pom
https://repo.scala-sbt.org/scalasbt/maven-snapshots/com/typesafe/sbt/sbt-site_2.12_1.0/1.4.0/sbt-site-1.4.0.pom
https://repo.scala-sbt.org/scalasbt/maven-snapshots/com/dwijnand/sbt-compat_2.12_1.0/1.2.6/sbt-compat-1.2.6.pom
https://repo.scala-sbt.org/scalasbt/maven-snapshots/com/typesafe/sbt/sbt-twirl_2.12_1.0/1.5.1/sbt-twirl-1.5.1.pom
https://repo.scala-sbt.org/scalasbt/maven-snapshots/io/gatling/gatling-sbt_2.12_1.0/3.1.0/gatling-sbt-3.1.0.pom
https://repo.scala-sbt.org/scalasbt/maven-snapshots/com/typesafe/sbt/sbt-web_2.12_1.0/1.4.4/sbt-web-1.4.4.pom
https://repo.scala-sbt.org/scalasbt/maven-snapshots/com/eed3si9n/sbt-unidoc_2.12_1.0/0.4.3/sbt-unidoc-0.4.3.pom
```

**Comment by SethTisue** [comment]
@vishalpawar17 to pick one off your list: https://repo.scala-sbt.org/scalasbt/maven-releases/com/typesafe/sbt/sbt-twirl_2.12_1.0/1.5.1/sbt-twirl-1.5.1.pom is available from https://repo.scala-sbt.org/scalasbt/sbt-plugin-releases , can you use that instead?

**Comment by eed3si9n** [comment]
@vishalpawar17 https://repo.scala-sbt.org/scalasbt/maven-releases was never used to serve sbt plugins. 404 is the correct response.

**Comment by eed3si9n** [comment]
I'm locking this issue because false information is just going to be confusing for readers who might need to read this issue in the future. In general, please see the migration status report that I wrote in https://eed3si9n.com/bintray-to-jfrog-artifactory-migration-status-and-sbt-1.5.1, and if you are having some issue regarding Artifactory, check for existing open issues or create a new issue please.

**Comment by xuwei-k** [comment]
- https://github.com/sbt/sbt/issues/7202 😢
